core/string/src/lib.rs

```diff
@@ -245,24 +304,35 @@ impl JsString {
                 // - The `RawJsString` type has all the necessary information to reconstruct a valid
                 //   slice (length and starting pointer).
                 //
-                // - We aligned `h.data` on allocation, and the block is of size `h.len`, so this
+                // - We aligned `h.data()` on allocation, and the block is of size `h.len`, so this
                 //   should only generate valid reads.
                 //
                 // - The lifetime of `&Self::Target` is shorter than the lifetime of `self`, as seen
                 //   by its signature, so this doesn't outlive `self`.
+                //
+                // - The `RawJsString` created from string literal has a static lifetime `JsStr`.
```

Comment:
Not sure what you mean by "static lifetime `JsStr`? If you mean `RawJsString` has a longer lifetime than `JsStr`, then you can put something like "A `RawJsString` created from a string literal has a static lifetime, making it always valid to reference using a `JsStr`"